core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -165,6 +176,63 @@ protected void beforeDisconnect() {
     // Empty implementation.
   }
 
+  /**
+   * Handshakes with meta master.
+   */
+  private synchronized void handshake() throws AlluxioStatusException {
+    if (isConnected() || sHandshakeComplete.get()) {
+      return;
+    }
+    mAddress = getAddress();
+    LOG.info("Alluxio client (version {}) is trying to bootstrap-connect with {}",
+        RuntimeConstants.VERSION, mAddress);
+    // A plain socket transport to bootstrap
+    TTransport baseTransport = ThriftUtils.createThriftSocket(mAddress);
+    TTransport transport = new BootstrapClientTransport(baseTransport);
+    TProtocol protocol = ThriftUtils.createThriftProtocol(transport,
+        Constants.META_MASTER_SERVICE_NAME);
+    List<ConfigProperty> clusterConfig;
+    try {
+      transport.open();
+      MetaMasterClientService.Client client = new MetaMasterClientService.Client(protocol);
+      clusterConfig = client.getConfiguration(new GetConfigurationTOptions())
+          .getConfigList()
+          .stream()
+          .map(ConfigProperty::fromThrift)
+          .collect(Collectors.toList());
+    } catch (TTransportException e) {
+      LOG.error("TTransportException", e);
+      throw new UnavailableException("TTransportException" + e, e);
+    } catch (TException e) {
+      LOG.error("TTransportException", e);
+      throw new UnavailableException("TransportException" + e, e);
+    } finally {
+      transport.close();
+    }
+    // merge conf returned by master as the cluster default into Configuration
+    Properties clusterProps = new Properties();
+    for (ConfigProperty property : clusterConfig) {
+      String name = property.getName();
+      if (PropertyKey.isValid(name) && property.getValue() != null) {
+        PropertyKey key = PropertyKey.fromString(name);
+        String value = property.getValue();
+        clusterProps.put(key, value);
+        LOG.debug("Loading cluster default: {} ({}) -> {}", key, key.getScope(), value);
+      }
+    }
+    String clientVersion = Configuration.get(PropertyKey.VERSION);
```

Comment:
the service version only needs to change when we change the thrift service in a way that is not backwards compatible. I don't think we've made changes like that since 1.4, so those constants should be correct.

Reply:
anyway, checking version during handshake is fine, but it should probably be info or debug level since we are supposed to support different minor and patch versions.